components/tidb_query_expr/src/impl_cast.rs

```diff
@@ -604,6 +606,25 @@ fn cast_any_as_string<T: ConvertTo<Bytes> + Evaluable + EvaluableRet>(
     }
 }
 
+#[rpn_fn(nullable, capture = [ctx, extra])]
+#[inline]
+fn cast_year_as_string(
+    ctx: &mut EvalContext,
+    extra: &RpnFnCallExtra,
+    val: Option<&Int>,
+) -> Result<Option<Bytes>> {
+    match val {
+        None => Ok(None),
+        Some(val) => {
+            let val = match *val as i64 {
```

Comment:
Maybe use `if` directly looks more clear?

Reply:
I want to try pattern matching, that's amazing!

Reply:
Pattern matching is cool, but it's too verbose here compared with:

```rust
let val = if *val == 0 { abc } else { def }
```